ring_describer_test.go

```diff
@@ -0,0 +1,376 @@
+		Table:    "local",
+		Name:     "host_id",
+		TypeInfo: NativeType{typ: TypeUUID},
+	}, {
+		Keyspace: "system",
+		Table:    "local",
+		Name:     "preferred_ip",
+		TypeInfo: NativeType{typ: TypeInet},
+	}, {
+		Keyspace: "system",
+		Table:    "local",
+		Name:     "rack",
+		TypeInfo: NativeType{typ: TypeVarchar},
+	}, {
+		Keyspace: "system",
+		Table:    "local",
+		Name:     "release_version",
+		TypeInfo: NativeType{typ: TypeVarchar},
+	}, {
+		Keyspace: "system",
+		Table:    "local",
+		Name:     "rpc_address",
+		TypeInfo: NativeType{typ: TypeInet},
+	}, {
+		Keyspace: "system",
+		Table:    "local",
+		Name:     "schema_version",
+		TypeInfo: NativeType{typ: TypeUUID},
+	}, {
+		Keyspace: "system",
+		Table:    "local",
+		Name:     "supported_features",
+		TypeInfo: NativeType{typ: TypeVarchar},
+	}, {
+		Keyspace: "system",
+		Table:    "local",
+		Name:     "tokens",
+		TypeInfo: CollectionType{
+			NativeType: NativeType{proto: 4, typ: TypeSet},
+			Elem:       NativeType{proto: 4, typ: TypeVarchar},
+		},
+	}},
+}
+
+func (*mockConnection) querySystem(ctx context.Context, query string) *Iter {
+	if query == "SELECT * FROM system.local WHERE key='local'" {
+		return &Iter{
+			meta:    systemLocalResultMetadata,
+			framer:  &mockFramer{data: framerLocalData},
+			numRows: 1,
+			next:    nil,
+		}
+	} else if query == "SELECT * FROM system.peers" {
+		return &Iter{
+			meta:    systemPeersResultMetadata,
+			framer:  &mockFramer{data: framerPeersData},
+			numRows: 2,
+			next:    nil,
+		}
+	}
+	return nil
+}
+
+func (*mockConnection) getIsSchemaV2() bool { return false }
+func (*mockConnection) setSchemaV2(s bool)  {}
+func (*mockConnection) query(ctx context.Context, statement string, values ...interface{}) (iter *Iter) {
+	return nil
+}
+func (*mockConnection) getScyllaSupported() scyllaSupported { return scyllaSupported{} }
+
+type mockControlConn struct{}
+
+func (m *mockControlConn) getConn() *connHost {
+	return &connHost{
+		conn: &mockConnection{},
+		host: &HostInfo{},
+	}
+}
+
+func (m *mockControlConn) awaitSchemaAgreement() error                                { return nil }
+func (m *mockControlConn) query(statement string, values ...interface{}) (iter *Iter) { return nil }
+func (m *mockControlConn) discoverProtocol(hosts []*HostInfo) (int, error)            { return 0, nil }
+func (m *mockControlConn) connect(hosts []*HostInfo) error                            { return nil }
+func (m *mockControlConn) close()                                                     {}
+func (m *mockControlConn) getSession() *Session                                       { return nil }
+
+func TestGetHosts(t *testing.T) {
+	r := &ringDescriber{control: &mockControlConn{}, cfg: &ClusterConfig{}}
+
+	hosts, _, err := r.GetHosts()
+	if err != nil {
+		t.Fatalf("unable to get hosts: %v", err)
+	}
+
+	// local host and one of the peers are zero token so only one peer should be returned
+	assertEqual(t, "hosts length", 1, len(hosts))
```

Comment:
What it it is returns zero-token host, can you please check exact data that supposed to be returned.